### PR TITLE
[NETBEANS-4635] - Add JakartaEE 8 Platform module

### PR DESCRIPTION
**Pull request by pepness**
This is an improvement to the [PR-2023](https://github.com/apache/netbeans/pull/2023) from Josh Juneau. I used the `j2ee.platform` module as a reference.

Notes:
- Add module `jakartaee8.platform` that contains the Javadoc
- Fix JakartaEE integration with the NetBeans library (Classpath & Javadoc)
  from both jakartaee-api & jakartaee-web-api libraries.

Classpath - Before
NetBeans-1
Classpath - After
NetBeans-1-after
Javadoc - Before
NetBeans-2
Javadoc - After
Javadoc-After

### CODE REVIEW

**Comment by neilcsmith-net** [comment]
Possible for 12.1?

@juneau001 please take a look. Can't seem to add you as a reviewer?!

**Comment by juneau001** [comment]
Hi Neil, I can take a look but timing at not work out.  I am away with my family and will have limited connectivity this week.   I will try to take a look at this, but probably cannot push any changes until early next week.

**Comment by neilcsmith-net** [comment]
Thanks @juneau001 @pepness 

Assuming retriggered tests are green, I'll merge into master for 12.1 beta2.

File: enterprise/jakartaee.platform/build.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        </condition>
+    </target>
+
+    <target name="prepare-doc" depends="-check-prepared-doc" unless="jakartaee.platform.doccreated">
+        <delete file="external/generated-jakarta.jakartaee-api-8.0.0-javadoc.jar" />
+
+        <!-- repackage doc file into the format netbeans used prior to switching to the maven central artefact -->
+        <jar destfile="external/generated-jakarta.jakartaee-api-8.0.0-javadoc.jar">
+            <zipfileset src="external/jakarta.jakartaee-api-8.0.0-javadoc.jar" includes="**" prefix="docs/api" />
+        </jar>
```
Comment (matthiasblaesing) [inline enterprise/jakartaee.platform/build.xml:45]: Please also add a license file for the generated package. See the `j2ee.platform/external` situation as reference. The generated package has the same license as its source, but we need the license information.

**Comment by matthiasblaesing** [comment]
@pepness  This is commit should do it: https://github.com/matthiasblaesing/netbeans/commit/a0cf9ea8300c761a4858ecae24651d4b94ba7299 - feel free to review and pull into this PR.

File: enterprise/jakartaee.platform/build.xml (hunk 1 above)
Comment (pepness) [reply enterprise/jakartaee.platform/build.xml:45]: Did not notice that license file, thanks.

**Comment by pepness** [comment]
@matthiasblaesing  I just review your PR it did the work done. I re-build and re-test this PR and also check that the binaries where in place.

**Comment by pepness** [comment]
> I don't know the history, so the question might be invalid: should the jakartaee.platform module contain a revision number? I would expect Jakarta 9 to change and they might (or even have?) decide to change packages, so will this impact us?
> 


You are right about this, JakartaEE 9 change the packages names from javax to jakarta:
- [Jakarta EE 9 API Docs](https://jakarta.ee/specifications/platform/9/apidocs/)
- [Jakarta EE 8 API Docs](https://jakarta.ee/specifications/platform/8/apidocs/)

This is the [Backwards Compatibility](https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan) statement for Jakarta EE 9

> 
> Backwards Compatibility
> 
> Jakarta EE 9 WILL NOT impose any backward compatibility requirements for compatible implementations to support the Jakarta EE 8 release. This is aligned with our goal of enabling new implementations to enter the ecosystem. However, we strongly believe that many products and tools will provide backwards compatibility and migration solutions for enabling older applications to run on Jakarta EE 9.

So I think the new module 'jakartaee.platform' should be renamed to 'jakartaee8.platform' and remove the revision number?

**Comment by matthiasblaesing** [comment]
Thank you for integrating and checking. For the version: Yes I think the name should contain the major version (that was what I meant with revision number in the name). If that is done, the doc artifact should also get the version number:

So `docs/jakartaee-doc-api.jar` should become `docs/jakartaee8-doc-api.jar` (feel free to reorder, the placement is just a suggestion).

**Comment by pepness** [comment]
Modification done. Update images and the description.

**Comment by neilcsmith-net** [comment]
Assuming review complete, this needs squashing and force pushing here.  @matthiasblaesing you happy with requested changes?  If not, have to decide whether to push back to 12.2.

On squashing, better if you can do it @pepness but if the branch permissions are open (as default) I can do that before merging.

**Comment by pepness** [comment]
Commits squashed.

**Comment by neilcsmith-net** [comment]
Thanks for handling the squash @pepness - merging now and I'll sync for beta 2.

**Pull request merged**